source/background/library/browserEvents.js

```diff
@@ -1,18 +1,28 @@
 import ms from "ms";
 import log from "../../shared/library/log.js";
 import { dispatch, getState } from "../redux/index.js";
-import { setAuthToken as setDropboxAuthToken } from "../../shared/actions/dropbox.js";
 import { setAuthCode as setGoogleDriveAuthCode } from "../../shared/actions/googleDrive.js";
 import { setUserActivity } from "../../shared/actions/app.js";
 import { getAutoLoginDetails } from "../../shared/selectors/autoLogin.js";
 import { clearAutoLogin } from "../../shared/actions/autoLogin.js";
 import { sendTabMessage } from "../../shared/library/extension.js";
 import { getEntry } from "./archives.js";
+import { setAuthToken as setDropboxAuthToken } from "../../shared/actions/dropbox.js";
+import {
+    getTokens as getMyButtercupTokens,
+    processNewVaultDetails as processMyButtercupVaultDetails
+} from "./myButtercup.js";
+import {
+    setAccessToken as setMyButtercupAccessToken,
+    setRefreshToken as setMyButtercupRefreshToken
+} from "../../shared/actions/myButtercup.js";
 
 const AUTOLOGIN_EXPIRY = ms("45s");
 const BUTTERCUP_DOMAIN_REXP = /^https:\/\/buttercup.pw\//;
 const DROPBOX_ACCESS_TOKEN_REXP = /access_token=([^&]+)/;
 const GOOGLE_DRIVE_AUTH_CODE_REXP = /\?googleauth&code=([^&#?]+)/;
+const MYBUTTERCUP_AUTH_CODE_REXP = /code=([^&]+)/;
+const MYBUTTERCUP_CALLBACK_URL_REXP = /^https?:\/\/(localhost:8000|my\.buttercup\.pw)\/oauth\/authorized\//;
```

Comment:
Is it safe to leave this `localhost` thing here?

Reply:
Yep! It's just for development. Validation of the redirect is performed by the server anyway.